ts/test/session/unit/disappearing_messages/DisappearingMessage_test.ts

```diff
@@ -0,0 +1,679 @@
+    it('returns undefined when disappearing messages is off', async () => {
+      const mode: DisappearingMessageConversationModeType = 'off';
+      const expirationStartTimestamp = setExpirationStartTimestamp(mode);
+
+      expect(expirationStartTimestamp, 'it should return undefined').to.be.undefined;
+    });
+    it('if we give it a timestamp it returns the older timestamp for deleteAfterRead', async () => {
+      const mode: DisappearingMessageConversationModeType = 'deleteAfterRead';
+      const timestamp = new Date().valueOf();
+      const expirationStartTimestamp = setExpirationStartTimestamp(mode, timestamp);
+
+      expect(expirationStartTimestamp, 'it should return a number').to.be.is.a('number');
+      expect(isValidUnixTimestamp(expirationStartTimestamp!), 'it should be a valid unix timestamp')
+        .to.be.true;
+      expect(
+        expirationStartTimestamp,
+        'expirationStartTimestamp should be less than the input timestamp'
+      ).to.be.lessThan(timestamp);
+    });
+    it('if we give it a timestamp it returns the older timestamp for deleteAfterSend', async () => {
+      const mode: DisappearingMessageConversationModeType = 'deleteAfterSend';
+      const timestamp = new Date().valueOf();
+      const expirationStartTimestamp = setExpirationStartTimestamp(mode, timestamp);
+
+      expect(expirationStartTimestamp, 'it should return a number').to.be.is.a('number');
+      expect(isValidUnixTimestamp(expirationStartTimestamp!), 'it should be a valid unix timestamp')
+        .to.be.true;
+      expect(
+        expirationStartTimestamp,
+        'expirationStartTimestamp should be less than the input timestamp'
+      ).to.be.lessThan(timestamp);
+    });
+    it('if we give it an invalid timestamp it returns undefined', async () => {
+      const mode: DisappearingMessageConversationModeType = 'deleteAfterSend';
+      const timestamp = -1;
+      const expirationStartTimestamp = setExpirationStartTimestamp(mode, timestamp);
+
+      expect(expirationStartTimestamp, 'it should return undefined').to.be.undefined;
+    });
+  });
+
+  describe('changeToDisappearingMessageType', () => {
+    it("if it's a Private Conversation and the expirationMode is off and expireTimer = 0 then the message's expirationType is unknown", async () => {
+      const conversation = new ConversationModel({
+        ...conversationArgs,
+      } as any);
+      const expireTimer = 0; // seconds
+      const expirationMode = 'off';
+      const messageExpirationType = changeToDisappearingMessageType(
+        conversation,
+        expireTimer,
+        expirationMode
+      );
+
+      expect(messageExpirationType, 'returns unknown').to.be.eq('unknown');
+    });
+    it("if it's a Private Conversation and the expirationMode is deleteAfterRead and expireTimer > 0 then the message's expirationType is deleteAfterRead", async () => {
+      const conversation = new ConversationModel({
+        ...conversationArgs,
+      } as any);
```

Comment:
we could name those as ConversationAttributes. Just because we want to  have the least amount of any in the app (and it would be nice to not have those when searching for : any)